src/repr/src/scalar.rs

```diff
@@ -606,6 +606,33 @@ impl<'a> Datum<'a> {
         }
         is_instance_of_scalar(self, &column_type.scalar_type)
     }
+
+    /// Returns the name of the type
+    pub fn type_name(&self) -> &'static str {
```

Comment:
Just adding a comment here so it's not missed, since it was added after the PR was approved.

Reply:
Ah, rather than this I was imagining simply using `"time".to_string()` in the error message.

This function is not as comprehensive as it seems because of e.g. custom types, which would need to return their specified names, which requires catalog access, which we don't have at this point. Additionally, complex types lose the declaration of their inner types, etc. We don't plan to support custom time-related types _and_ time is simple, though, so I don't think there's much risk in including the type name in the error message above. However, given that this isn't actually generalizable, don't think an apparently-general solution is quite right.

Welcome to the wonders of SQL :D

Reply:
Got it. I was originally nervous about hardcoding the type because while `date_part_time_inner` is only accessible for the `Time` type, other functions like `date_part_timestamp_inner` are accessible through multiple types. Though since for now the error is only returned in places were the type is certain, it's probably not a big deal.